torch/_prims_common/__init__.py

```diff
@@ -48,8 +48,8 @@ def __rmul__(self, other: Any) -> typing.Self:
     _IntLikeT = TypeVar("_IntLikeT", bound=_WorksWithInt)
 
 
-ShapeType: TypeAlias = Union[torch.Size, list[int], tuple[int, ...]]
-StrideType: TypeAlias = Union[list[int], tuple[int, ...]]
+ShapeType: TypeAlias = Sequence[int]
```

Comment:
Still not sure about this part... could be too permissive for these type aliases. 

Can I get another reviewer to take a look?

Reply:
Sure, I'd be flattered by another reviewer!

My reasoning is that we never use `int` as a type for _data,_ like the contents of a tensor, but only for _metadata_ like size, strides and array indices.

A `Tensor` is a `Sequence[Tensor]` and a `np.ndarray` might be a `Sequence[np.int64]` but never a `Sequence[int]`.

Reply:
Okay, I'm pretty sure reducing this to a Sequence is bad in this case after reviewing similar changes to DimsType. It is a bit more restrictive but also purposefully restricted user containers and other potential abc sequences.

Reply:
Sure, I'll revert that and amend or ignore elsewhere.

A `torch.Size` isa `tuple[int, ...]`, should I remove that from the list of types as redundant?

Reply:
I think it's better to think of it a place holder should we want to make torch.Size more experssive later on.